feat: thank you page

olehrakwix opened the pull request.

VladyslavG commented:
how about mobile screen size ?

yurii-ve commented on app/routes/thank-you/route.tsx:
```diff
@@ -13,14 +17,29 @@ export default function ThankYouPage() {
+                <div className={styles.paragraph}>
+                    <div>You will receive a confirmation email soon.</div>
+                    <br />
+                    {order ? <div>Your order number: {order.number}</div> : null}
```
`{order && <div>...</div>}`?

yurii-ve commented on app/routes/thank-you/route.tsx:
```diff
@@ -13,14 +17,29 @@ export default function ThankYouPage() {
-                </p>
+                <div className={styles.paragraph}>
+                    <div>You will receive a confirmation email soon.</div>
+                    <br />
```
why not add some gap with css?

yurii-ve commented on src/components/order-summary/line-item/line-item.tsx:
```diff
@@ -0,0 +1,74 @@
+
+const DEFAULT_IMAGE_QUALITY = 90;
+
+export function getImageUrl(imageUrl: string, options?: ImageUrlOptions) {
```
let's move it out of the component file.. put in the `utils.ts` in the component folder

yurii-ve commented on src/components/order-summary/line-item/line-item.tsx:
```diff
@@ -0,0 +1,74 @@
+}
+
+export const LineItem = ({ item }: LineItemProps) => {
+    const lineItemImage = item.image
```
`lineItemImageUrl`

yurii-ve commented on src/components/order-summary/line-item/line-item.tsx:
```diff
@@ -0,0 +1,74 @@
+        <div className={styles.root}>
+            <div className={styles.productInfo}>
+                <div>
+                    {lineItemImage ? (
```
`{lineItemImage && ...}`

yurii-ve commented on src/components/order-summary/line-item/line-item.tsx:
```diff
@@ -0,0 +1,74 @@
+                <div className={styles.productDetails}>
+                    <div className={styles.productName}>{item.productName?.translated}</div>
+                    <div>{item.price?.formattedAmount}</div>
+                    {item.descriptionLines?.map((l) => (
```
`l` => `descriptionLine`

yurii-ve commented on _codux/boards/components/order-summery/order-summery.board.tsx:
```diff
@@ -0,0 +1,8 @@
+
+export default createBoard({
+    name: 'OrderSummary',
+    Board: () => <OrderSummary order={{}} />,
```
why does this board takes an empty object order?

yurii-ve commented on src/components/order-summary/order-summary.tsx:
```diff
@@ -0,0 +1,112 @@
+                    ))}
+                </div>
+
+                <hr className={styles.divider} />
```
`hr` doesn't fit here
```
The <hr> HTML element represents a thematic break between paragraph-level elements: for example, a change of scene in a story, or a shift of topic within a section.
```

yurii-ve commented on src/components/order-summary/order-summary.tsx:
```diff
@@ -0,0 +1,112 @@
+        <div className={styles.root}>
+            <div className={cx(styles.section)}>
+                <div className={styles.lineItems}>
+                    {order.lineItems?.map((i) => (
```
`i` -> `lintItem`

yurii-ve commented on src/components/order-summary/order-summary.tsx:
```diff
@@ -0,0 +1,112 @@
+                <hr className={styles.divider} />
+
+                <div className={styles.priceDetailsWrapper}>
+                    <div />
```
?

yurii-ve commented on src/components/order-summary/order-summary.tsx:
```diff
@@ -0,0 +1,112 @@
+                    <h6 className={styles.title}>Delivery address</h6>
+                    <ul className={styles.addressLines}>
+                        <li>
+                            {`${order?.shippingInfo?.logistics?.shippingDestination?.contactDetails?.firstName} ${order?.shippingInfo?.logistics?.shippingDestination?.contactDetails?.lastName}`}
```
it's not readable.. let's destructure `firstName` and `lastName` somewhere in the component body

yurii-ve commented on src/components/order-summary/order-summary.tsx:
```diff
@@ -0,0 +1,112 @@
+    );
+};
+
+function renderAddress(address: orders.Address | undefined) {
```
let's do this in the component body

yurii-ve commented on src/components/order-summary/order-summary.tsx:
```diff
@@ -0,0 +1,112 @@
+}
+
+/**
+ * This component was created using Codux's Default new component template.
```
remove this comment

alisaLachman commented:
@olehrakwix 

looks great, just one small adjustment, align the prices more to the right:

Before:
<img width="708" alt="Screenshot 2024-09-10 at 12 18 28" src="https://github.com/user-attachments/assets/3e4ee82f-99c5-4b06-9d0e-6c1d0adc0a00">
After:
<img width="653" alt="Screenshot 2024-09-10 at 12 21 07" src="https://github.com/user-attachments/assets/1919b440-dfb0-4316-bbc0-d8afd56529ac">
<img width="655" alt="Screenshot 2024-09-10 at 12 21 18" src="https://github.com/user-attachments/assets/6d7e5d28-dd9d-448e-b9c0-01d45466c5c7">



@ShireeWill 
is anything else is missing regarding the information in the thank you page?

olehrakwix commented:
@alisaLachman , addressed comments, please see updated view in description.
regarding buyer note: if API will return it - it will be shown, but I have never seen it in real flow
